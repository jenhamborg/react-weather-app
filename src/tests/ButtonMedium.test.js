import React from "react";
import { render, cleanup } from "@testing-library/react";
import ButtonMedium from "../components/reusable/mediumButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer.create(<ButtonMedium />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("Medium Button", () => {
  afterEach(() => {
    cleanup;
  });

  it("Correct text is displayed", async () => {
    const mockFunction = jest.fn();
    const { getByText } = render(
      <ButtonMedium onClick={mockFunction} text="New York" />
    );

    const button = getByText("New York");

    expect(button).toHaveTextContent("New York");
  });
});
