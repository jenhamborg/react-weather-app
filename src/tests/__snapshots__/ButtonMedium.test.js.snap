// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="mediumButton__Button-sc-1e5cium-0 dScmLB"
/>
`;
